pkg/descheduler/leaderelection.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package descheduler
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/uuid"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	componentbaseconfig "k8s.io/component-base/config"
+	"k8s.io/klog/v2"
+	"os"
+)
+
+// NewLeaderElection starts the leader election code loop
+func NewLeaderElection(
+	run func() error,
+	client clientset.Interface,
+	LeaderElectionConfig *componentbaseconfig.LeaderElectionConfiguration,
+	ctx context.Context,
+) error {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("unable to get hostname: %v", err)
+	}
+
+	id := fmt.Sprintf("%s-%s", hostname, uuid.New().String())
```

Comment:
```go
import "k8s.io/apimachinery/pkg/util/uuid"
...
if hostname, err := os.Hostname(); err != nil {
	// on errors, make sure we're unique
	id = string(uuid.NewUUID())
} else {
	// add a uniquifier so that two processes on the same host don't accidentally both become active
	id = hostname + "_" + string(uuid.NewUUID())
}
```